songpal/main.py

```diff
@@ -227,20 +227,33 @@ def print_settings(settings, depth=0):
 
 
 @cli.command()
+@click.option("--output", type=str, required=False)
 @click.argument("input", required=False)
 @pass_dev
 @coro
-async def input(dev: Device, input):
+async def input(dev: Device, input, output):
     """Get and change outputs."""
     inputs = await dev.get_inputs()
     if input:
         click.echo("Activating %s" % input)
         try:
             input = next((x for x in inputs if x.title == input))
-            await input.activate()
         except StopIteration:
             click.echo("Unable to find input %s" % input)
             return
+        zone = None
+        if output:
+            zones = await dev.get_zones()
+            try:
+                zone = next((x for x in zones if x.title == output))
+            except StopIteration:
```

Comment:
This all feels very fragile; I suppose it is fine for now, but I think at some point (after/if I get ever time to do refactoring), such permanent properties will be cached by the main device class.

Reply:
Agreed! See comment above on the `Volume` print string as well. I'd be happy to help with that, but didn't want to make such a big change to your repository without some input/checking

Reply:
After some thought, how about adding a helper called `get_zone(name)` which returns the zone if available, or raises an exception (SongpalException) when not?

The same should be done for `activate(zone)` when the input is not valid.

Reply:
Added that helper to `Device` and replaced code for the `zone` `volume` and `input` CLI commands